### PR TITLE
New feature: Allow parallel deployment of multiple hosts

### LINKED ISSUE #123: Re-prompt in case the incorrect answer
**Problem**: through the deployment of Blockscout instance the script asks several times awaiting `Yes` or `No` answers. In case of another answer, it continues with the default option without prompting again.

**Solution**: re-prompt in case the incorrect answer instead of continuing with the default option

### LINKED ISSUE #124: Parallelize BS deployment
The idea is to update script to allow parallelized deployment of multiple BlockScout instances.
Points:
- [ ] Make new version backward compatible to current
- [x] Ensure running multiple BlockScout instances on the same VM are not affecting each other.
- [x] Make possibility to update networks with multiple prefixes.

### LINKED ISSUE #125: Add new variables
https://forum.poa.network/t/faq-blockscout-environment-variables/1814

Screenshot 2019-05-22 at 11 54 44

### PR DESCRIPTION
**Pull request by ArseniiPetrovich**
closes #124, closes #125.

This PR allows user to deploy multiple BlockScout instances in parallel. User should specify the host list and host/group vars and run the deployment. Each instance will be launched in a separate host port to avoid collisions.

Feature list:
- [x] Parallel deployment
- [x] Deployment at multiple hosts
- [x] Re-prompt in case of incorrect answer

TODO list before merge:
- [ ] This is a breaking change, so we should probably tag the current version as the 1.0 release before merging this commit. 
- [x] There is still a problem with SUPPORTED_CHAINS variable. It works either at dev or production environment depending on syntax.
- [x] Improve README to make the variable assignment clear;
- [x] Test deployment from scratch at least couple of times;

### CODE REVIEW

**Comment by ArseniiPetrovich** [comment]
- [x] Tested the single deployment. Works well. Multi-deploy need to be tested as well before merging.

File: README.md [hunk 1]
```diff
@@ -142,37 +133,41 @@ The configuration variable `db_storage` can be used to define the amount of stor
...
+Where `[group]` is a group name, which will be interpreted as a `prefix` for all created resources and `host` is a name of BlockScout instance.
 
-5. Run `ansible-playbook deploy_infra.yml`; 
+4. For each host merge `infrastructure.yml.example` and `all.yml.example` config template files in `host_vars` folder into single config file with the same name as in `hosts` file:
 
-   - During the deployment the ["diffs didn't match"](#error-applying-plan-diffs-didnt-match) error may occur, it will be ignored automatically. If  Ansible play recap shows 0 failed plays, then the deployment was successful despite the error.
+```bash
+cat host_vars/infrastructure.yml.example host_vars/all.yml.example > host_vars/host.yml
+```
+5. For each group merge `infrastructure.yml.example` and `all.yml.example` config template files in `group_vars` folder into single config file with the same name as group name in `hosts` file:
```
Comment (vbaranov) [inline README.md]: 5. and 6. are not from the new line

File: README.md [hunk 2]
```diff
@@ -142,37 +133,41 @@ The configuration variable `db_storage` can be used to define the amount of stor
...
+```ini
+[group]
+host
 ```
 
-4. Set the variables at `group_vars/all.yml` config template file as described at the [corresponding part of instruction](#Configuration);
+Where `[group]` is a group name, which will be interpreted as a `prefix` for all created resources and `host` is a name of BlockScout instance.
 
-5. Run `ansible-playbook deploy_infra.yml`; 
+4. For each host merge `infrastructure.yml.example` and `all.yml.example` config template files in `host_vars` folder into single config file with the same name as in `hosts` file:
```
Comment (vbaranov) [inline README.md]: with the same name of what?

File: group_vars/all.yml.example [hunk 3]
```diff
@@ -19,50 +19,3 @@ upload_debug_info_to_s3: true
 ## Name of the bucket where TF state files will be stored
 bucket: "poa-terraform-state"
 
-## All resources will be prefixed with this one
-prefix: "poa"
```
Comment (vbaranov) [inline group_vars/all.yml.example]: @ArseniiPetrovich where is prefix now configured?

Comment (ArseniiPetrovich) [reply group_vars/all.yml.example]: Now the prefix is the group name you configure at hosts file ;)

File: README.md (hunk 2 above)
Comment (ArseniiPetrovich) [reply README.md]: As the same name you named your host in a hosts file.
For example:
[first]
host1
host2

[second]
host3
host4

In this case we will need to create `host1.yml`, `host2.yml`, `host3.yml`, `host4.yml` files.

File: group_vars/all.yml.example (hunk 3 above)
Comment (vbaranov) [reply group_vars/all.yml.example]: where this is indicated in a new Readme?

File: README.md (hunk 2 above)
Comment (vbaranov) [reply README.md]: @ArseniiPetrovich I understand this. But this is not obvious for an average user. Please add an example to README

**Comment by vbaranov** [comment]
@ArseniiPetrovich why has `do not merge` label been applied to this PR?

**Comment by ArseniiPetrovich** [comment]
> @ArseniiPetrovich why has `do not merge` label been applied to this PR?

Well... It was added to prevent from merging while PR was in "WIP" state.
Luckily, I forgot to remove it.
As for now we can't merge this PR because I've found a bug with a way Ansible merge the hashes when they are specified both in group and host vars. It occurred, that if we specify the empty array in a host_vars - it will never be merged with group_vars. 
I think we should either wait while the solution for the bug will be found, or make a note about the bug.

Please, see the related issue for details => https://github.com/ansible/ansible/issues/58733

File: group_vars/all.yml.example (hunk 3 above)
Comment (ArseniiPetrovich) [reply group_vars/all.yml.example]: Quote from README - lines 139-146:
```
The simplest `hosts` file with one BlockScout instance will look like:

[group]
host

Where `[group]` is a group name, which will be interpreted as a `prefix` for all created resources and `host` is a name of BlockScout instance.
```

File: README.md (hunk 2 above)
Comment (ArseniiPetrovich) [reply README.md]: Added.

**Comment by ArseniiPetrovich** [comment]
@vbaranov 
What do you think if we merge this PR with a side note about empty hashes? 
I'm not sure how long would it take to fix the Ansible issue.

**Comment by vbaranov** [comment]
>@ArseniiPetrovich @symanovich-a
5. and 6. paragraphs are not from the beginning of the line

Any updates on that?

**Comment by symanovich-a** [comment]
@vbaranov check, please

**Pull request merged**